### PR TITLE
Predicate Estimation `fuel-core` part

### PR DESCRIPTION
**Pull request by ghost**
The PR adds support for predicate estimation from [`fuel-vm`](https://github.com/FuelLabs/fuel-vm/pull/386).

The change adds:
- All predicate inputs now accept the `predicate_gas_used` field.
- New API endpoint `estimate_predicates` that estimates predicates and returns the transaction with updated predicates.

### CODE REVIEW

File: bin/fuel-core-client/src/main.rs [hunk 1]
```diff
@@ -15,6 +15,8 @@ enum TransactionCommands {
     /// Submit a JSON encoded transaction for inclusion in a block
     Submit { tx: String },
     /// Submit a JSON encoded transaction for a dry-run execution
+    Estimate { tx: String },
```
Comment (Voxelot) [inline bin/fuel-core-client/src/main.rs:18]:
```suggestion
    EstimatePredicates { tx: String },
```

we don't want to confuse people with thinking this is a general estimation API like what the SDK does.

File: bin/fuel-core-client/src/main.rs [hunk 2]
```diff
@@ -15,6 +15,8 @@ enum TransactionCommands {
     /// Submit a JSON encoded transaction for inclusion in a block
     Submit { tx: String },
     /// Submit a JSON encoded transaction for a dry-run execution
```
Comment (Voxelot) [inline bin/fuel-core-client/src/main.rs:17]: This doesn't use dry-run at all right?

File: crates/fuel-core/src/schema/tx/input.rs [hunk 3]
```diff
@@ -174,6 +184,7 @@ impl From<&fuel_tx::Input> for Input {
                 tx_pointer: TxPointer(*tx_pointer),
                 witness_index: *witness_index,
                 maturity: (*maturity).into(),
+                predicate_gas_used: 0.into(),
```
Comment (Voxelot) [inline crates/fuel-core/src/schema/tx/input.rs:187]: Should this use the value from the input?

**Review by Voxelot — APPROVED** [review]

**Review by Voxelot — APPROVED** [review]
approving - but I don't think this change is mergable until the VM is released.

**Comment by xgreenx** [comment]
Close in favor of the https://github.com/FuelLabs/fuel-core/pull/1195

**Pull request closed without merging**